CMakeLists.txt

```diff
@@ -98,6 +98,8 @@ if(MSVC AND FREECAD_LIBPACK_USE AND LIBPACK_FOUND)
 endif()
 
 if (BUILD_TEST)
+    include(CTest)
+    enable_testing()
```

Comment:
reason for these? I've got no clue on how the whole gtest thing works

Reply:
basically:
- tests are added with the gtest library.
- an executable called `Tests_run` is created. Upon running this all the tests created will get evaluated and a report is printed
- this is not integrated with the IDE
- the two lines you mentioned enable ctest. It acts as a middleman between gtest and the IDE.